### PR TITLE
Fix: Upward scroll fling value not consumed when refresh callback is called

### PR DESCRIPTION
**Pull request by LakeLab**
This PR is my revision of #15. 
I would appreciate feedback from the maintainers.

### CODE REVIEW

**Comment by LakeLab** [comment]
@wingio, Could you help me check whether this is acceptable?

**Comment by wingio** [comment]
> @wingio, Could you help me check whether this is acceptable?

yeah i could do that tomorrow, and im sorry for not getting to it for so long, i wasn't aware this was opened

**Comment by LakeLab** [comment]
For your information, I can revert the code for the sample module. I just wanted to show you the case which is #15 issue.

**Comment by LakeLab** [comment]
> yeah i could do that tomorrow, and im sorry for not getting to it for so long, i wasn't aware this was opened

No problem! Thank you!

**Comment by LakeLab** [comment]
@wingio 
pleaseee

**Review by wingio — APPROVED** [review]

**Comment by wingio** [comment]
LGTM

File: demo/src/main/java/dev/materii/pullrefresh/demo/sample/DragRefreshSample.kt [hunk 1]
```diff
@@ -31,13 +34,22 @@ fun DragRefreshSample(
             )
         }
     ) {
-        Box(
-            modifier = Modifier
-                .fillMaxSize()
-                .verticalScroll(rememberScrollState()),
-            contentAlignment = Alignment.Center
+        LazyColumn(
+            Modifier.fillMaxSize(),
```
Comment (X1nto) [inline demo/src/main/java/dev/materii/pullrefresh/demo/sample/DragRefreshSample.kt:38]: We use named arguments
```suggestion
            modifier = Modifier.fillMaxSize(),
```

File: demo/src/main/java/dev/materii/pullrefresh/demo/sample/DragRefreshSample.kt [hunk 2]
```diff
@@ -31,13 +34,22 @@ fun DragRefreshSample(
...
+            items(100) {
+                Text(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(5.dp)
+                        .background(
+                            color = MaterialTheme.colorScheme.outline,
+                            shape = RoundedCornerShape(10.dp)
+                        )
+                        .padding(10.dp), text = "No. $it"
```
Comment (X1nto) [inline demo/src/main/java/dev/materii/pullrefresh/demo/sample/DragRefreshSample.kt:50]: These shouldn't be on the same line

File: pullrefresh/src/commonMain/kotlin/dev/materii/pullrefresh/PullRefreshState.kt [hunk 3]
```diff
@@ -125,17 +125,23 @@ class PullRefreshState internal constructor(
...
         }
         animateIndicatorTo(0f)
         val consumed = when {
             // We are flinging without having dragged the pull refresh (for example a fling inside
             // a list) - don't consume
             distancePulled == 0f -> 0f
-            // If the velocity is negative, the fling is upwards, and we don't want to prevent the
-            // the list from scrolling
-            velocity < 0f -> 0f
+            velocity < 0f -> if (isRefreshTiming) {
```
Comment (X1nto) [inline pullrefresh/src/commonMain/kotlin/dev/materii/pullrefresh/PullRefreshState.kt:137]: A one-liner arrow for a multiline if seems weird, use blocks instead
```suggestion
            velocity < 0f -> { 
            if (isRefreshTiming) {
```

File: pullrefresh/src/commonMain/kotlin/dev/materii/pullrefresh/PullRefreshState.kt [hunk 4]
```diff
@@ -125,17 +125,23 @@ class PullRefreshState internal constructor(
...
+                // We need to prevent the fling upward when the refresh starts.
+                velocity
+            }else {
```
Comment (X1nto) [inline pullrefresh/src/commonMain/kotlin/dev/materii/pullrefresh/PullRefreshState.kt:140]: Whitespace
```suggestion
            } else {
```

**Comment by X1nto** [comment]
Oh crap I had this tab open like 6 hours ago, forgot to press "Request changes". @wingio could you review code style too next time?

**Pull request merged**